src/models/generated/ColocatedSupplyStorageTechnology.jl

```diff
@@ -0,0 +1,353 @@
+- `operation_costs_wind::PSY.OperationalCost`: (default: `ThermalGenerationCost()`) Fixed and variable O&M costs for a technology
+- `balancing_topology::String`: Set of balancing nodes
+- `efficiency_storage::InOut`: (default: `(in=1, out=1)`) Efficiency of charging storage
+- `region::Vector{Region}`: (default: `Vector()`) Zone where tech operates in
+- `losses_storage::Float64`: (default: `1.0`) Power loss (pct per hour)
+- `inverter_supply_ratio::Float64`: Ratio of generation capacity to grid connection capacity
+- `ext::Dict`: (default: `Dict()`) Option for providing additional data
+- `capital_costs_wind::PSY.ValueCurve`: (default: `LinearCurve(0.0)`) Capital costs for investing in a technology.
+- `internal::InfrastructureSystemsInternal`: (default: `InfrastructureSystemsInternal()`) Internal field
+- `lifetime_solar::Int`: (default: `100`) Maximum number of years a technology can be active once installed
+- `base_year::Int`: (default: `2020`) Reference year for technology data
+- `existing_capacity_energy::Float64`: (default: `0.0`) Pre-existing energy capacity for a storage technology (MWh)
+- `capital_costs_inverter::PSY.ValueCurve`: Capitals costs for investing in inverter capacity
+- `max_inverter_capacity::Float64`: (default: `1e8`) Limit on inverter capacity (MW)
+- `operations_costs_energy::PSY.OperationalCost`: (default: `StorageCost()`) Fixed and variable O&M costs for a storage technology
+- `financial_data_storage::TechnologyFinancialData`: Struct containing relevant financial information for a technology
+- `capital_costs_solar::PSY.ValueCurve`: (default: `LinearCurve(0.0)`) Capital costs for investing in a technology.
+"""
+mutable struct ColocatedSupplyStorageTechnology{T <: PSY.Generator} <: Technology
+    "Maximum number of years a technology can be active once installed"
+    lifetime_storage::Int
+    "identifies whether the technology is available"
+    available::Bool
+    "Fixed and variable O&M costs for a technology"
+    operation_costs_solar::PSY.OperationalCost
+    "Maximum allowable installed capacity for a technology"
+    capacity_limits_wind::MinMax
+    "The technology name"
+    name::String
+    "Capital costs for investing in a storage technology's charge/discharge capacity."
+    capital_costs_power::PSY.ValueCurve
+    "allowable installed power capacity for a storage technology"
+    capacity_power_limits::MinMax
+    "Pre-existing capacity for a technology"
+    initial_capacity_wind::Float64
+    "Maximum number of years a technology can be active once installed"
+    lifetime_wind::Int
+    "Fixed and variable O&M costs for a storage technology"
+    operations_costs_power::PSY.OperationalCost
+    "allowable installed energy capacity for a storage technology"
+    capacity_energy_limits::MinMax
+    "ID for individual generator"
+    id::Int64
+    "Minimum required durection for a storage technology"
```

Comment:
I guess it should say duration? Its funny, because its a combination of duration and direction lol